code/modules/food_and_drinks/food.dm

```diff
@@ -19,13 +19,18 @@
 	if(last_check_time + 50 < world.time)
 		if(ishuman(M))
 			var/mob/living/carbon/human/H = M
-			if(foodtype & H.dna.species.toxic_food)
-				to_chat(H,"<span class='warning'>What the hell was that thing?!</span>")
-				H.adjust_disgust(25 + 30 * fraction)
-			else if(foodtype & H.dna.species.disliked_food)
-				to_chat(H,"<span class='notice'>That didn't taste very good...</span>")
-				H.adjust_disgust(11 + 15 * fraction)
-			else if(foodtype & H.dna.species.liked_food)
-				to_chat(H,"<span class='notice'>I love this taste!</span>")
-				H.adjust_disgust(-5 + -2.5 * fraction)
+			if(!H.has_trait(TRAIT_AGEUSIA))
+				if(foodtype & H.dna.species.toxic_food)
+					to_chat(H,"<span class='warning'>What the hell was that thing?!</span>")
+					H.adjust_disgust(25 + 30 * fraction)
```

Comment:
I think the "food like/dislike" system mixes both preferences and intolerance a lot.

Moths should not be able to digest meat effectively etc.

Reply:
If you want I can refactor food likes/dislikes, when I originally made it I was new to byond code

Reply:
I'm thinking of giving it a go.